### DISCUSSION TITLE
Question: Custom channels

### DISCUSSION THREAD

**Discussion by saddy001** [neutrinolabs/xrdp#2202] [Q&A] [answered]
Hello,

I'd like to catch a link click event within the client session and catch that in the RDP client guacd (to ultimately open a link in the "outer" webbrowser). See https://issues.apache.org/jira/browse/GUACAMOLE-1563

Does xrdp support a simple way of adding custom channels? Or would it need to be implemented directly in the core?

**Comment by matt335672** [comment, marked as answer]
There is a way, but it's not well documented.

There's an example in https://github.com/neutrinolabs/xrdp/blob/devel/xrdpapi/simple.c which shows a way to use the echo virtual channel documented in [[MS-RDPEECO]](https://docs.microsoft.com/en-us/openspecs/windows_protocols/ms-rdpeeco/dd36d1eb-2e97-4b24-b4a3-0ca68d500521)

**Comment by saddy001** [comment]
Thanks. The code says "THIS TEST WILL FAIL IF CONNECTED FROM A NON WINDOWS RDP CLIENT". Does it mean that it is Windows-only?

**Comment by matt335672** [comment]
I think it just means that at the time it was written, only mstsc.exe supported the echo virtual channel.
